### PR TITLE
 Added a Camera control switcher

### PR DESCRIPTION
**Pull request by FreeFrags**
I am sorry i messed up i closed my other one too (#7148) as i had to pull dev first

Modified each Control

moved addEventListeners to a function "addEventListeners"
added a removeEventListeners function
I ran into this issue:
http://stackoverflow.com/questions/32412507/switch-between-flycontrols-and-orbitcontrols-threejs-mouse-clicked?noredirect=1#comment52768065_32412507

### CODE REVIEW

**Comment by mrdoob** [comment]
I think the name `addEventListeners` is confusing... I think `enable` and `disable` may be better...

**Comment by FreeFrags** [comment]
Ok I will change the name, it ll be sometime tomorrow I think.

**Comment by FreeFrags** [comment]
Renamed as you pointed out.

**Comment by WestLangley** [comment]
I see that `OrbitControls` now has `this.enabled` and `this.enable()`. Also `this.dispose()` which is reusing a method name used elsewhere.

How do you envision the use case for these methods?

File: examples/js/controls/ControlSwitcher.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+function ControlSwitcher(fov, aspect, near, far, container, onCameraChanged) {
```
Comment (WestLangley) [inline examples/js/controls/ControlSwitcher.js]: If you would like help with your formatting, see the Wiki article [Mr.doob's-Code-Style™](https://github.com/mrdoob/three.js/wiki/Mr.doob%27s-Code-Style™).

**Comment by FreeFrags** [comment]
I will rename the functions again I think enableEventlisteners and disableEventlisteners might be a better name for the functions then.

The use case for these methods: 
The eventlisteners which are not removed by its self (some are like eg added on mousedown and removed on mouseup) should be added to these functions. This way when you call this function before the delete the cameracamera control the eventlisteners are properly removed.

I ran into an issue when switching from orbit to fly controls where the eventlisteners were not removed. so thats why i propose that the controls get a common function which can be called to remove them before a delete.

Im not sure if it is possible in JS but ideally i would like to see this function be called in a destructor so when you call delete or the object goes out of scope the function is automatically called.

http://stackoverflow.com/questions/32412507/switch-between-flycontrols-and-orbitcontrols-threejs-mouse-clicked?noredirect=1#comment52768065_32412507

As for the formatting i will take a look at the coding style and correct it as much as possible.

**Comment by FreeFrags** [comment]
@WestLangley i changed the function names. 

About the this.dispose function it was already there, so i called the disableEventlisteners from there since i dont know where in the rest of the code it is used and didnt want to break that. 

I hope the code has now been formatted correctly.

**Comment by threejsworker** [comment]
The examples of this pullrequest are now built and visible in threejsworker. To view them, go to the following link: 

 http://threejsworker.com/viewpullrequest.html#7151

**Comment by FreeFrags** [comment]
@threejsworker followed the link however i cant seem to view them.

**Pull request closed without merging**